pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriter.java

```diff
@@ -143,6 +153,10 @@ public TxnLogBufferedWriter(ManagedLedger managedLedger, OrderedExecutor ordered
                                 DataSerializer<T> dataSerializer,
                                 int batchedWriteMaxRecords, int batchedWriteMaxSize, int batchedWriteMaxDelayInMillis,
                                 boolean batchEnabled, TxnLogBufferedWriterMetricsStats metrics){
+        if (batchedWriteMaxRecords <= 1 && batchEnabled){
+            log.warn("The current maximum number of records per batch is set to 1. Disabling Batch will improve"
```

Comment:
Perhaps "Transaction Log Buffered Writer has batching enabled yet the maximum batch size was configured to less than or equal to 1 record, hence due to performance reasons batching is disabled"

Could it be they wouldn't know per that log line which configuration to change since this can be used in Pending Ack Store and TxLog?

Reply:
Thinking out loud - @codelipenghui - do you think it's better to do this type of validation and change when processing configuration?

Reply:
> Thinking out loud - @codelipenghui - do you think it's better to do this type of validation and change when processing configuration?

Yes, I support having a validation when processing configuration. E.g. batchedWriteMaxRecords >= 10, size >= 128k

Reply:
Hi @asafm @codelipenghui 

Can we add validation in the following PR, the validation will trigger test rewrite, and there is already a plan to rewrite `TxnLogBufferedWriterTest.testMainProcess`, I will add validation in that PR